init.go

```diff
@@ -27,6 +27,9 @@ func initTlsPemPair(certFile, keyFile string, clientConfig *rest.Config, client
 	if certFile != "" || keyFile != "" {
 		tlsPair = tlsPairFromFiles(certFile, keyFile)
 	}
+	// if cert & key defined in secret(tls.kyverno) use it,
```

Comment:
Something very strange happened here:
1. The certs loaded from files are ignored
2. TlsPair does pretty the same what does tlsPairFromCluster after it
You have justa added your own code without considering the existing functionality. Now the code is incomprehensible and will have to be rewritten.

We **shoud** use the same tls secret for self-signed or not self-signed certificate, because the source of certificate makes no difference for webhook server, it is just a pair of files. You should check the presence of only CA bundle, and make decisions from the fact of its presence. If the CA bundle is in a cluster, we use it together with a pair of keys written to the secret that was used before (you can change the type of the old secret to tls). If there is no user CA bundle, we use a bundle from kubconfig, or vice versa.

BTW, from now, we can remove certFile and keyFile parameters from the controller's command line arguments, because we can use in-cluster secrets even if controller is running outside the cluster. This can be done in scope of another task.
FYI @JimBugwadia